stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -1631,9 +1631,13 @@ extension ${Self} {
   public init?(exactly value: ${That}) {
     _value = Builtin.${sign}itofp_${ThatBuiltinName}_FPIEEE${bits}(value._value)
 
-%   if srcBits >= SignificandBitCount:
-    guard let roundTrip = ${That}(exactly: self),
-      roundTrip == value else {
+%   if srcBits > SignificandBitCount:
+    // The magnitude of the integer must be representable by the significand.
+    // Omit trailing zeros as they will be encoded by the exponent.
+    let absv = value.magnitude
+    let mask: ${That}.Magnitude = ~((1 << (${SignificandBitCount}+1)) - 1)
```

Comment:
I'm mainly concerned about the point I called out above in light of this; specifically, I'm worried about the case where `SignificandBitCount+1 == srcBits`.